app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,225 @@
+                ->first();
+
+            if ($defaultProductSizePrice) {
+                $defaultSize['price'] = $defaultProductSizePrice->price;
+            }
+        }
+        $nameDisabled = OrderLine::where('product_id', $productId)->exists();
+        return view('admin.editProduct', [
+            'product' => $product,
+            'baseCategories' => $categories,
+            'baseGroups' => $groups,
+            'baseProductSizes' => $productSizes,
+            'baseChosenCategorie' => $chosenCategorie,
+            'chosenGroups' => $chosenGroups,
+            'sizesWithPrices' => $sizes,
+            'defaultSizeWithPrice' => $defaultSize,
+            'nameDisabled' => $nameDisabled,
+        ]);
+    }
+    public function store(ProductCreationRequest $request)
+    {
+        $requestData = $request->all();
+        $image = $request->file('af-submit-app-upload-images');
+        $imageName = $request->input('name');
+        $uniqueImageName = $imageName . '_' . time() . '.' . $image->getClientOriginalExtension();
+        $product = Product::create([
+            'name' => $imageName,
+            'image_path' => $this->savePicture($image, $uniqueImageName),
+            'product_type_id' => $this->categoryToId($requestData['category']),
+            'discount' => 0,
+        ]);
+        $defaultPriceForSize = ['Default' => null];
+        $priceForSize = $requestData['priceForSize'] ?? $defaultPriceForSize;
+        $customSizes = $requestData['custom_sizes'] ?? [];
+        $customPrices = $requestData['custom_prices'] ?? [];
+        $sizesAndPrices = array_merge($priceForSize, array_combine($customSizes, $customPrices));
+        foreach ($sizesAndPrices as $size => $price) {
+            if ($price !== null) {
+                $productSize = ProductSize::firstOrCreate(['size' => $size]);
+                $product->productSizes()->syncWithoutDetaching([$productSize->id => ['price' => $price]]);
+            }
+        }
+        $groups = $requestData['products-group-multiselect'] ?? [];
+        foreach ($groups as $group) {
+            $product->groups()->attach(Group::firstOrCreate(['name' => $group]));
+        }
+        $request->session()->flash('toast-type', 'success');
+        $request->session()->flash('toast-message', __('toast/messages.success-product-add'));
+        return redirect()->route('manage.products.index');
+    }
+    private function savePicture($picture, $name)
+    {
+        if (!$picture) {
+            return "/images/products/placeholder.png";
+        }
+        if (Storage::disk('public')->put('/images/products/' . $name . '.png', $picture->get())) {
+            return '/images/products/' . $name . '.png';
+        }
+        return "/images/products/placeholder.png";
+    }
+    private function categoryToId($category)
+    {
+        $category = strtolower($category);
+        $productType = ProductType::firstOrCreate(['type' => $category]);
```

Comment:
Wil je hier wel een create doen? Ik zou juist een error throwen

Reply:
Hiermee maken we gelijk nieuwe product typen aan, in dit geval kleuren. Anders moet je alle kleuren van tevoren vastellen.
Zoiets doen we ook met de product sizes